### ISSUE TITLE
After pull to latest composerised version, problems with autoloader

### ISSUE THREAD

**Issue by mbirth** [spotweb/spotweb#201]
@GeoffreyDijkstra After I pulled the latest commit, it was missing folders like boxcar/boxcar from the vendor directory. It looks like you committed several composer modules as submodules without having a `.gitmodules` file. So my git doesn't know where to get it.

I'm pretty sure you meant to delete the `.git` folders inside the modules in `vendor` and commit them directly.

**Comment by GeoffreyDijkstra** [comment]
I think I don't entirely follow you right now... The vendor folder is committed but if we would remove it and run composer install again we shouldn't miss files. If this is the case then those missing files should be added to the corresponding packages. So can you elaborate where the .gitmodules should be used for?

**Comment by mbirth** [comment]
Have a look at this:

![2016-11-04 17_39_46-spotweb_vendor_boxcar at master spotweb_spotweb](https://cloud.githubusercontent.com/assets/130302/20014059/baebfe2e-a2b5-11e6-97a5-77ac326b5875.png)

When checking out spotweb, the git client will only create an empty directory `boxcar`.

Because there are `.git` directories (created by composer) on your harddisk, your git client only committed the composer module's base folder as an empty folder and marked it as a git submodule.

**Comment by GeoffreyDijkstra** [comment]
Ah I see, this is one of the reasons I never include the vendor dir (bad practice). Can you fix this?

**Comment by mbirth** [comment]
Okay, I'll fix this later tonight (still at work, but leaving soon).

**Comment by mesa57** [comment]
@GeoffreyDijkstra : please revert the composer pull as this breaks spotweb totally.

**Comment by GeoffreyDijkstra** [comment]
@mesa57 This is the one  problem I didn't hink about and is caused by including the vendor folder.

People that now how to use GIT also know how to use composer. The other part that want full zip files we can provide those through the releases section of github.

But @mbirth is going to fix this and I'm already looking at it to, so give us a sec.

**Comment by mbirth** [comment]
@GeoffreyDijkstra In case you want to fix it yourself: Search for `.git` directories below the `vendor` folder and delete those. Then commit all the files. I've restored the branch in my repo, so just commit to that. Then you should be able to merge the changes and everything should be fine.

I'll head home now and see how far you got. :)

**Comment by mesa57** [comment]
You over estimate the knowlegde level of git of many users.
The only command they know is pull, and that;s it.

**Comment by mbirth** [comment]
By the way: What about some Slack room or Gitter.im ? Those would be a great thing to coordinate development - especially in a case like this.

**Comment by GeoffreyDijkstra** [comment]
@mbirth @mesa57 I fixed it, but because you reverted I had to force push so I didn't have to fix a lot of merge conflicts. Everything went fine, so as you can see now we got a working vendor folder. 🎉

**Comment by mesa57** [comment]
Question : is client.php on nntp overwriten with a new version ? We did some essential patching there ....

**Comment by GeoffreyDijkstra** [comment]
@mesa57 No I copied the Full NNTP directory from spotweb to that package, so it is the same version.
I also updated the 1.0.0 tag: https://github.com/spotweb/spotweb/releases/tag/1.0.0
I add you to that repo. But atm @mbirth and I got both packages that are spotweb specific, those should be moved to the spotweb (company) github and also to a spotweb packagist account. (so packages will eventually look like spotweb/nntp, spotweb/thingy, spotweb/... and spotweb/spotweb as project ofcourse).
